scripts/assemble_changelog.py

```diff
@@ -448,14 +453,23 @@ def list_files_to_merge(options):
     files_to_merge.sort(key=EntryFileSortKey)
     return files_to_merge
 
+def check_extensions(options):
+    files = glob.glob(os.path.join(options.dir, '*'))
+    files = {x for x in files if not x.endswith(".txt")}
+    files.discard("ChangeLog.d/00README.md")
+    if files:
+        raise FilePathError(files)
+
 def merge_entries(options):
     """Merge changelog entries into the changelog file.
 
     Read the changelog file from options.input.
+    Check that all entries have a .txt extension
     Read entries to merge from the directory options.dir.
     Write the new changelog to options.output.
     Remove the merged entries if options.keep_entries is false.
     """
+    check_extensions(options)
```

Comment:
The implementation of this check is closely tied to the implementation of `list_files_to_merge`, so `check_extensions` should be called from (or even inlined into) `list_files_to_merge`.

Reply:
Or could we just make `list_files_to_merge` throw the newly introduced exception?